tests/bdd/features/volume/resize/test_resize_offline.py

```diff
@@ -0,0 +1,308 @@
+def all_the_replicas_of_volume_should_be_resized_to_new_capacity():
+    """all the replicas of volume should be resized to new capacity."""
+    volume = ApiClient.volumes_api().get_volume(VOLUME_UUID)
+    replicas = volume.state.replica_topology
+    for replica_uuid in replicas:
+        replica = ApiClient.replicas_api().get_replica(replica_uuid)
+        assert replica.size >= VOLUME_NEW_SIZE
+
+
+@then("the failure reason should be invalid arguments")
+def the_failure_reason_should_be_invalid_arguments():
+    """the failure reason should be invalid arguments."""
+    # Reason code: Invalid Arguments, mapped from
+    # SvcError::VolumeResizeArgsInvalid
+    assert pytest.exception.status == 400
+
+
+@then("the failure reason should be volume-in-use precondition")
+def the_failure_reason_should_be_volumeinuse_precondition():
+    """the failure reason should be volume-in-use precondition."""
+    # Reason code: Conflict, mapped from SvcError::InUse
+    assert pytest.exception.status == 409
+
+
+@then("the new capacity should be available for the application to use")
+def the_new_capacity_should_be_available_for_the_application_to_use():
+    """the new capacity should be available for the application to use."""
+    volume = ApiClient.volumes_api().get_volume(VOLUME_UUID)
+    assert volume.state.target["size"] >= VOLUME_NEW_SIZE
+
+
+@then("the volume expand status should be failure")
+def the_volume_expand_status_should_be_failure():
+    """the volume expand status should be failure."""
+    the_volume_resize_status_should_be_failure()
+
+
+@then("the volume expand status should be success")
+def the_volume_expand_status_should_be_success():
+    """the volume expand status should be success."""
+    volume = ApiClient.volumes_api().get_volume(VOLUME_UUID)
+    assert (
+        volume.spec.size == VOLUME_NEW_SIZE
+    )  # Can we do Operation spec.operation check for None?
+
+
+@then("the volume is expanded to the new capacity")
+def the_volume_is_expanded_to_the_new_capacity():
+    """the volume is expanded to the new capacity."""
+    volume = ApiClient.volumes_api().get_volume(VOLUME_UUID)
+    assert volume.spec.size == VOLUME_NEW_SIZE
+    # spec.operation is cleared when volume resize completes.
+    assert hasattr(volume.spec, "operation") is False
+
+
+@then("the volume resize status should be failure")
+def the_volume_resize_status_should_be_failure():
+    """the volume resize status should be failure."""
+    assert pytest.exception is not None
+
+
+@then("the volume should get published with expanded capacity")
+def the_volume_should_get_published_with_expanded_capacity():
+    """the volume should get published with expanded capacity."""
+    volume = ApiClient.volumes_api().get_volume(VOLUME_UUID)
+    assert hasattr(volume.spec, "target")
+    assert str(volume.spec.target.protocol) == str(Protocol("nvmf"))
+    assert volume.spec.size == VOLUME_NEW_SIZE
+
+
+def del_volume():
```

Comment:
you can use new api, which helps debug things on failure:
`Volume.cleanup(volume)`